development/ts-migration-dashboard/files-to-convert.json

```diff
@@ -0,0 +1,1737 @@
+[
```

Comment:
This file holds all of the files that we want to convert to TypeScript. It is generated from the [file patterns defined in `development/ts-migration-dashboard/scripts/constants.ts`](https://github.com/MetaMask/metamask-extension/pull/13820/files#diff-7403df2bdf0769e46840036cc85a72d36a7254adcf04e68be71ca9288ea1dd11R4), which includes the same entrypoints that the build system uses as well as patterns to represent tests and Storybook files. The files matched by the patterns are used as starting points and the dependency graphs of these files are traversed via the [`madge`](https://github.com/pahen/madge) package to produce this file list.

Why this approach? Because if we were to just use a glob like `{app,development,shared,ui}/**/*.js` then we might catch files that don't end up being used anywhere. By traversing the dependency graph we are essentially guaranteeing that all the files listed here are being referred to somewhere.

Why commit this file to the codebase? Because as we convert files to TypeScript, we may end up introducing new files, and we don't want those files to appear in the dashboard. By maintaining a list of the original files to convert we can ensure that the files listed on the dashboard don't change.